stac_validator/stac_validator.py

```diff
@@ -0,0 +1,372 @@
+        geojson_key = "geojson_resolver"
+        item_key = "item-{}".format(self.stac_version)
+        catalog_key = "catalog-{}".format(self.stac_version)
+
+        if item_key in cache and catalog_key in cache:
+            logging.debug('Using STAC specs from local cache.')
+            self.geojson_resolver = RefResolver(
+                base_uri="file://{}/".format(self.dirpath), referrer="geojson.json"
+            )
+            return cache[item_key], cache[geojson_key], cache[catalog_key]
+
+        else:
+            try:
+                logging.debug('Gathering STAC specs from remote.')
+                stac_item_geojson = requests.get(
+                    StacVersion.item_geojson_schema_url(version)
+                ).json()
+                stac_item = requests.get(StacVersion.item_schema_url(version)).json()
+                stac_catalog = requests.get(StacVersion.catalog_schema_url(version)).json()
+            except Exception as error:
+                logger.exception("STAC Download Error")
+                raise VersionException(f"Could not download STAC specification files for version: {version}")
+
+        self.dirpath = tempfile.mkdtemp()
+
+        with open(os.path.join(self.dirpath, "geojson.json"), "w") as fp:
+            logging.debug("Copying GeoJSON spec from local file to cache")
+            geojson_schema = json.dumps(stac_item_geojson)
+            fp.write(geojson_schema)
+            cache[geojson_key] = self.dirpath
+            self.geojson_resolver = RefResolver(
+                base_uri="file://{}/".format(self.dirpath), referrer="geojson.json"
+            )
+
+        stac_item_file = StacVersion.fix_stac_item(version, "stac-item.json")
+
+        with open(os.path.join(self.dirpath, stac_item_file), "w") as fp:
+            logging.debug("Copying STAC item spec from local file to cache")
+            stac_item_schema = json.dumps(stac_item)
+            fp.write(stac_item_schema)
+            cache[item_key] = stac_item_schema
+
+        with open(os.path.join(self.dirpath, "stac-catalog.json"), "w") as fp:
+            logging.debug("Copying STAC catalog spec from local file to cache")
+            stac_catalog_schema = json.dumps(stac_catalog)
+            fp.write(stac_catalog_schema)
+            cache[catalog_key] = stac_catalog_schema
+
+        ITEM_SCHEMA = os.path.join(self.dirpath, "stac-item.json")
+        ITEM_GEOJSON_SCHEMA = os.path.join(self.dirpath, "geojson.json")
+        CATALOG_SCHEMA = os.path.join(self.dirpath, "stac-catalog.json")
+
+        return ITEM_SCHEMA, ITEM_GEOJSON_SCHEMA, CATALOG_SCHEMA
+
+    def validate_json(self, stac_content, schema):
+        """
+        Validate stac
+        :param stac_content: input stac file content
+        :param schema of STAC (item, catalog)
+        :return: validation message
+        """
+
+        stac_schema = json.loads(schema)
+        try:
+            if 'title' in stac_schema and 'item' in stac_schema['title'].lower():
```

Comment:
single quotes, has this been run through black?

Reply:
fixed